packages/core/src/sync-realtime/index.ts

```diff
@@ -513,32 +582,59 @@ export const createRealtimeSync = (
     // Traces
     ////////
 
-    const shouldRequestTraces = callTraceFilters.length > 0;
+    const shouldRequestTraces =
+      transactionFilters.length > 0 || transferFilters.length > 0;
 
-    let callTraces: SyncCallTrace[] = [];
+    let traces: SyncTrace[] = [];
     if (shouldRequestTraces) {
-      const traces = await _trace_block(args.requestQueue, {
+      traces = await _debug_traceBlockByNumber(args.requestQueue, {
```

Comment:
Should use `_debug_traceBlockByHash`. It's more robust when reorgs occur because it unambiguously refers to the exact block